hadoop-tools/hadoop-aws/src/main/java/org/apache/hadoop/fs/s3a/S3AFileStatus.java

```diff
@@ -32,18 +32,24 @@
 @InterfaceStability.Evolving
 public class S3AFileStatus extends FileStatus {
   private boolean isEmptyDirectory;
+  private final String owner;
```

Comment:
I think we can achieve this change without adding a member variable in the subclass.  (See more specific notes to follow.)  Removing this member variable would reduce memory footprint.  Admittedly, the memory cost is probably not significant, but as we start thinking about the possibility of caching `FileStatus` instances client-side for things like S3Guard, then the per-instance memory cost of each `FileStatus` could become more significant.